web/src/search_pill.ts

```diff
@@ -147,22 +148,83 @@ function append_user_pill(
     operator: string,
     negated: boolean,
 ): void {
-    const pill_data: SearchUserPill = {
-        type: "search_user",
-        operator,
-        negated,
-        users: users.map((user) => ({
-            full_name: user.full_name,
-            user_id: user.user_id,
-            email: user.email,
-            img_src: people.small_avatar_url_for_person(user),
-            status_emoji_info: user_status.get_status_emoji(user.user_id),
-            should_add_guest_user_indicator: people.should_add_guest_user_indicator(user.user_id),
-            deactivated: !people.is_person_active(user.user_id) && !user.is_inaccessible_user,
-        })),
-    };
```

Comment:
I'm not sure why you needed to replace this fairly short block of code with 3x as much -- does this have behavior changes for the `dm` pills?

Assuming this is the right final code, I think it'd be helpful if you could reorganize this PR with a series of no-functional-changes refactors that are easy to verify don't modify the `dm` code.